### PR TITLE
Bearings support and control

### LINKED ISSUE #193: Add support for Mapbox Directions API's `bearings` option
See the description: https://docs.mapbox.com/api/navigation/directions/#:~:text=mapbox/driving.-,bearings,-integer

Need to figure out a way to introduce respective functionality into the plugin's UI.

### PR DESCRIPTION
**Pull request by smellyshovel**
This PR:
1. Adds the `bearings` configuration option.
2. Adds support for waypoints' bearings: public getter/setter `waypointsBearings`. These have no effect with disabled `bearings` - see i. 1.
3. Adds a new `rotatewaypoints` event which is emited each time new bearings values are set using the setter from i. 1.
4. Updates the `buildRequest` method to include the bearings info when making a request (when `bearings` is enabled).
5. Adds new `BearingsControl` control.
6. Adds an example that shows the new functionality in action.
7. Closes #193 

And 2 unrelated to this PR small changes, see below.

### CODE REVIEW

File: src/directions/main.ts [hunk 1]
```diff
@@ -147,7 +151,7 @@ export default class MapLibreGlDirections extends MapLibreGlDirectionsEvented {
         if (response.code !== "Ok") throw new Error(response.message ?? "An unexpected error occurred.");
       } finally {
         // see #189 (https://github.com/maplibre/maplibre-gl-directions/issues/189)
-        if (this.abortController.signal.reason !== "DESTROY") this.interactive = prevInteractive;
+        if (this.abortController?.signal.reason !== "DESTROY") this.interactive = prevInteractive;
```
Comment (smellyshovel) [inline src/directions/main.ts:154]: Another fix, should've been a separate PR, but I decided to include it here not to waste the time.

File: src/directions/main.ts [hunk 2]
```diff
@@ -755,7 +759,11 @@ export default class MapLibreGlDirections extends MapLibreGlDirectionsEvented {
   protected assignWaypointsCategories() {
     this._waypoints.forEach((waypoint, index) => {
       const category = index === 0 ? "ORIGIN" : index === this._waypoints.length - 1 ? "DESTINATION" : undefined;
-      if (waypoint.properties) waypoint.properties.category = category;
+
+      if (waypoint.properties) {
+        waypoint.properties.index = index;
```
Comment (smellyshovel) [inline src/directions/main.ts:764]: Another separate-PR-candidate. Basically, just saves a waypoint's index (for each of the waypoints) in their Feature Properties object (allows to e.g. display number of each waypoint as symbol (text) on the map).

**Comment by smellyshovel** [comment]
@Miguel-Sanches please, review this PR.
Tell me about anything that's missing.

Would be very nice if you could test it manually locally. Especially the Demo.

File: package.json [hunk 3]
```diff
@@ -70,8 +70,8 @@
     "@types/mapbox__point-geometry": "^0.1.2",
     "@types/mapbox__polyline": "^1.0.2",
     "@types/mapbox__vector-tile": "^1.3.0",
-    "@typescript-eslint/parser": "^5.46.1",
     "@typescript-eslint/eslint-plugin": "^5.46.1",
+    "@typescript-eslint/parser": "^5.46.1",
```
Comment (smellyshovel) [inline package.json:74]: Ignore... Probably the autoformatter's exhaust.

**Review by Miguel-Sanches — APPROVED** [review]

**Pull request merged**